packages/quantic/force-app/main/default/lwc/quanticDidYouMean/e2e/quanticDidYouMean.e2e.ts

```diff
@@ -0,0 +1,237 @@
+import {useCaseTestCases} from '../../../../../../playwright/utils/useCase';
+import {testInsight, testSearch, expect} from './fixture';
+import mockData from './data';
+
+const {
+  originalQuery: expectedOriginalQuery,
+  correctedQuery: expectedCorrectedQuery,
+  didYouMeanData,
+  didYouMeanNextData,
+  queryTriggerData,
+  triggeredQuery: expectedTriggeredQuery,
+} = mockData;
+
+const fixtures = {
+  search: testSearch,
+  insight: testInsight,
+};
+
+useCaseTestCases.forEach((useCase) => {
+  let test = fixtures[useCase.value];
+
+  test.describe(`quantic did you mean ${useCase.label}`, () => {
+    test.describe('when we have an automatic did you mean query correction', () => {
+      test.describe('with queryCorrectionMode set to "legacy"', () => {
+        test.use({
+          didYouMeanData,
+          options: {
+            disableQueryAutoCorrection: false,
+            queryCorrectionMode: 'legacy',
+          },
+        });
+        test('should display the didYouMean component and send the analytics', async ({
```

Comment:
here why are we checking the displaying of the component instead of inside unit tests? I thought we agreed to test those usecases in unit tests. Unless I am missing something and for a specific reason we needed to check that in the e2e?

Reply:
I still like to have a baseline "does the component render correctly" done as a e2e test, could potentially catch any browser related limitation, for example an issue with Locker/LWS preventing the component to render correctly.
I'm not sure Jest could correctly catch issues related to Locker.

And it does check that it sends the correct analytics too.